### ISSUE TITLE
taxonomic inference only populating most frequent kingdoms

### ISSUE THREAD

**Issue by sformel-usgs** [gbif/ipt#2372]
Yesterday we published this resource: https://ipt-obis.gbif.us/resource.do?r=shea_boehm_2024

It contains occurrences in 5 kingdoms, [as shown on gbif](https://www.gbif.org/dataset/9358fbd7-cfd0-4eab-99fa-0934396a0529/metrics). But the taxonomic inference tool only found three kingdoms: Chromista, Plantae, Animalia

One unusual thing we did in publishing this dataset is that we first uploaded metadata output from the eDNA tool( https://edna-tool.gbif-uat.org/).  So we didn't start from a completely blank slate.

**Comment by mike-podolskiy90** [comment]
Yes, there is a problem there. We have a limit on the maximum number of inferred taxa of 200, but it hits that limit early because it is not counting records correctly.

I've fixed the counting issue:
![Screenshot 2024-03-14 at 21 42 19](https://github.com/gbif/ipt/assets/24795387/962f17b5-e351-44e5-a4de-6a371ae0e83c)

but still has only 3 kingdoms because of the limit. So I'm thinking about these changes:

- Change limit functionality: apply it to each level separately (kingdom, phylum, class, order, family)
- Display an alert or any other message, that when the limit reached

What do you think?

**Comment by sformel-usgs** [comment]
I think your two bullets are good solutions. 

**Applying different limits for each taxonomic level**
Related to this discussion of downstream sharing of metadata to ODIS https://github.com/iodepo/odis-arch/issues/391 I would like to hear if @pieterprovoost has any thoughts on what appropriate limiting for each level.  

I think it's important to get complete lists of kingdom and phyla. Datasets that strongly focus on specific lower levels are less likely to be hyper-diverse, so it's probably ok if incomplete lists of family, order and class are occasionally encountered.

**Comment by mike-podolskiy90** [comment]
We just need to find reasonable numbers for the limits

**Comment by pieterprovoost** [comment]
Would it make sense to only show levels which have n < limit taxa? So list all phyla, then all classes etc, as long as the lists do not hit the limit. Showing incomplete lists is a bit misleading.

**Comment by sformel-usgs** [comment]
Good point that incomplete lists are misleading. I think Pieter's suggestion is a good one. 

It occurred to me that variation in the # of levels recorded might frustrate people who are searching across datasets with tools like ODIS. But...since many people don't include this metadata, or already include it to varying levels, it probably shouldn't be a big concern.
 
Separately, I noticed in the revision above that it's counting `NA` as a value, this can be excluded too.

**Comment by mike-podolskiy90** [comment]
@sformel-usgs Do you think to display something like this would be okay:

![Screenshot 2024-03-18 at 13 25 47](https://github.com/gbif/ipt/assets/24795387/3e0dc856-6d90-4b09-b7b6-ae07fb6fd831)

**Comment by sformel-usgs** [comment]
@mike-podolskiy90 that looks perfect, it's very easy to understand.

**Comment by mike-podolskiy90** [comment]
Thank you.

You also mentioned filtering. `NA` is clearly should be removed, but what are other values we should consider removing?
This kingdom looks odd `µl` by the way

**Comment by mike-podolskiy90** [comment]
I'm now concerned about value filtering - it might not be good performance-wise

**Comment by sformel-usgs** [comment]
Thanks for wrapping this up. Good catch on the kingdom `µl`. There is no microliter kingdom :-) That's an error in the data.  I'll follow up with the data provider.

**Comment by sformel-usgs** [comment]
@mike-podolskiy90 actually upon closer inspection, I don't see `µl` as a value in Kingdom for this data.  Any thoughts on where it might be coming from?

**Comment by mike-podolskiy90** [comment]
That's strange, I'll have a look

**Comment by sformel-usgs** [comment]
Actually, I just 're-inferred' and got a DNA sequences in there now: 

![image](https://github.com/gbif/ipt/assets/118295402/fb93e7cd-ad98-40b2-b7a9-847be4164813)

**Comment by mike-podolskiy90** [comment]
Where does it come from? That's something very unusual. Looking into it

**Comment by mike-podolskiy90** [comment]
I found the issue. IPT incorrectly analyses some fields that it shouldn't. It happens when there are multiple mappings.

**Comment by mike-podolskiy90** [comment]
I've created an issue #2403
